war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -694,9 +694,12 @@ var jenkinsRules = {
     "INPUT.required" : function(e) { registerRegexpValidator(e,/./,"Field is required"); },
 
 // validate form values to be an integer
-    "INPUT.number" : function(e) { registerRegexpValidator(e,/^(\d+|)$/,"Not an integer"); },
+    "INPUT.number" : function(e) { registerRegexpValidator(e,/^\-?(\d+)$/,"Not an integer"); },
+    "INPUT.non-negative-number" : function(e) {
+        registerRegexpValidator(e,/^\d+$/,"Not a non-negative number");
+    },
     "INPUT.positive-number" : function(e) {
-        registerRegexpValidator(e,/^(\d*[1-9]\d*|)$/,"Not a positive integer");
+        registerRegexpValidator(e,/[1-9]\d*/,"Not a positive integer");
```

Comment:
This needs to be anchored with `^…$`, as it seems JavaScript `String.match(RegExp)` searches for the first match, like Java’s `find` but unlike Java’s `matches`.